src/prefect/projects/base.py

```diff
@@ -0,0 +1,226 @@
+    """
+    path = Path(".").resolve()
+    parent = path.parent.resolve()
+    while path != parent:
+        prefect_dir = path.joinpath(".prefect")
+        if prefect_dir.is_dir():
+            return prefect_dir
+
+        path = parent.resolve()
+        parent = path.parent.resolve()
+
+
+def set_default_deployment_yaml(path: str) -> bool:
+    """
+    Creates default deployment.yaml file in the provided path if one does not already exist;
+    returns boolean specifying whether a file was created.
+    """
+    path = Path(path)
+    if (path / "deployment.yaml").exists():
+        return False
+    default_file = Path(__file__).parent / "templates" / "deployment.yaml"
+    with open(path / "deployment.yaml", "w") as f:
+        f.write(default_file.read_text())
+    return True
+
+
+def set_prefect_hidden_dir() -> bool:
+    """
+    Creates default .prefect directory if one does not already exist.
+    Returns boolean specifying whether a directory was created.
+    """
+    path = Path(".") / ".prefect"
+
+    # use exists so that we dont accidentally overwrite a file
+    if path.exists():
+        return False
+    path.mkdir()
+    return True
+
+
+def set_default_project_yaml(
+    path: str, name: str = None, pull_step: dict = None
+) -> bool:
+    """
+    Creates default prefect.yaml file in the provided path if one does not already exist;
+    returns boolean specifying whether a file was created.
+    """
+    path = Path(path)
+    if (path / "prefect.yaml").exists():
+        return False
+    default_file = Path(__file__).parent / "templates" / "prefect.yaml"
+    with open(default_file, "r") as df:
+        contents = yaml.safe_load(df)
+
+    import prefect
+
+    contents["prefect-version"] = prefect.__version__
+    contents["name"] = name
+
+    with open(path / "prefect.yaml", "w") as f:
+        # write header
+        f.write(
+            "# File for configuring project / deployment build, push and pull steps\n\n"
+        )
+
+        f.write("# Generic metadata about this project\n")
+        yaml.dump({"name": contents["name"]}, f, sort_keys=False)
+        yaml.dump({"prefect-version": contents["prefect-version"]}, f, sort_keys=False)
+        f.write("\n")
+
+        # build
+        f.write("# build section allows you to manage and build docker images\n")
+        yaml.dump({"build": contents["build"]}, f, sort_keys=False)
+        f.write("\n")
+
+        # push
+        f.write(
+            "# push section allows you to manage if and how this project is uploaded to"
+            " remote locations\n"
+        )
+        yaml.dump({"push": contents["push"]}, f, sort_keys=False)
+        f.write("\n")
+
+        # pull
+        f.write(
+            "# pull section allows you to provide instructions for cloning this project"
+            " in remote locations\n"
+        )
+        yaml.dump({"pull": pull_step or contents["pull"]}, f, sort_keys=False)
```

Comment:
This confused me at first, but it looks like we have to write the `prefect.yaml` file this way to include comments. In the future, we should consider using a templating solution for maintainability.